statsmodels/base/_screening.py

```diff
@@ -0,0 +1,214 @@
+                                                skip_hessian=True)
+
+            keep = np.abs(res_pen.params) > self.threshold_trim
+            # use largest params to keep
+            if len(res_pen.params) > self.k_max_included:
+                # TODO we can use now np.partition with partial sort
+                thresh_params = np.sort(np.abs(res_pen.params))[-self.k_max_included]
+                keep2 = np.abs(res_pen.params) > thresh_params
+                keep = np.logical_and(keep, keep2)
+            idx_nonzero = idx[keep]
+            # TODO: problem with order between using mask and indexing
+            #idx_nonzero.sort()
+
+            if disp:
+                print(idx_nonzero)
+            x0 = x[:, idx_nonzero]
+            start_params = res_pen.params[keep]
+
+            # use mask to get excluded
+            mask_excl = np.ones(k_vars, dtype=bool)
+            mask_excl[idx_nonzero] = False
+            idx_excl = np.nonzero(mask_excl)[0]
+            x1 = x[:, idx_excl]
+            history['idx_nonzero'].append(idx_nonzero)
+            history['keep'].append(keep)
+            history['params_keep'].append(start_params)
+
+        # final esimate
+        res_final = model_class(endog, x[:, idx_nonzero], penal=self.penal,
+                                pen_weight=nobs * 10,
+                                **self.init_kwds).fit(method='bfgs',
+                                          start_params=res_pen.params[keep],
+                                          warn_convergence=False, disp=disp)
+
+        res = ScreeningResults(self,
+                               results_pen = res_pen,
+                               results_final = res_final,
+                               idx_nonzero = idx_nonzero,
+                               idx_excl = idx_excl,
+                               start_params = start_params,
+                               history = history)
+        return res
+
+    def screen_exog_iterator(self, exog_iterator):
+        res_batches = []
+        res_idx = []
+        exog_winner = []
+        exog_idx = []
+        for ex in exog_iterator:
+            res_screen = self.screen_exog(ex, maxiter=20)
+            res_batches.append(res_screen)
+            res_idx.append(res_screen.idx_nonzero)
+            exog_winner.append(ex[:, res_screen.idx_nonzero[1:] - 1])
+            exog_idx.append(res_screen.idx_nonzero[1:] - 1)
+
+        exog_winner = np.column_stack(exog_winner)
+        res_screen_final = self.screen_exog(exog_winner, maxiter=20)
+
+
+        exog_winner_names = ['var%d_%d' % (bidx, idx)
+                             for bidx, batch in enumerate(exog_idx)
+                             for idx in batch]
+
+        idx_full = [(bidx, idx)
+                    for bidx, batch in enumerate(exog_idx)
+                    for idx in batch]
+        ex_final_idx = res_screen_final.idx_nonzero[1:] - 1
```

Comment:
the -1 here hard codes that `exog_keep` is constant only in my example, same above line 193, 194

Reply:
there is still a TODO for only indexing exog_candidates and not exog_keep